IceFishing/AppDelegate.swift

```diff
@@ -189,4 +239,81 @@ class AppDelegate: UIResponder, UIApplicationDelegate, SWRevealViewControllerDel
 			revealController.frontViewController.view.userInteractionEnabled = false
 		}
 	}
+	
+	func applicationDidBecomeActive(application: UIApplication) {
+		if #available(iOS 9.0, *) {
+			guard let shortcut = launchedShortcutItem else { return }
+			
+			if FBSession.activeSession().isOpen {
+				handleShortcutItem(shortcut as! UIApplicationShortcutItem)
+				launchedShortcutItem = nil
+			}
+		}
+	}
+	
+	// MARK: - Force Touch Shortcut
+	
+	@available(iOS 9.0, *)
+	func application(application: UIApplication, performActionForShortcutItem shortcutItem: UIApplicationShortcutItem, completionHandler: (Bool) -> Void) {
+		let handleShortcutItem = self.handleShortcutItem(shortcutItem)
+		completionHandler(handleShortcutItem)
+	}
+	
+	enum ShortcutIdentifier: String {
+		case Post
+		case PeopleSearch
+		case Liked
+		case Profile
+		
+		init?(fullType: String) {
+			guard let last = fullType.componentsSeparatedByString(".").last else {return nil}
+			self.init(rawValue: last)
+		}
+		
+		var type: String {
+			return NSBundle.mainBundle().bundleIdentifier! + ".\(self.rawValue)"
+		}
+	}
+	
+	@available(iOS 9.0, *)
+	func handleShortcutItem(shortcutItem: UIApplicationShortcutItem) -> Bool {
+		guard ShortcutIdentifier(fullType: shortcutItem.type) != nil else { return false }
+		guard let shortcutType = shortcutItem.type as String? else { return false }
+
+		func handleShortCutForMenuIndex(index: Int) {
+			var vc: UIViewController!
+			if index == -1 {
+				vc = profileVC
+			} else {
+				vc = sidebarVC.elements[index].viewController
+			}
+			revealVC.setFrontViewPosition(.Left, animated: false)
+			navigationController.setViewControllers([vc], animated: false)
+			sidebarVC.preselectedIndex = index
+		}
+		
+		switch (shortcutType) {
+		case ShortcutIdentifier.Post.type:
+			//Bring up Search for Post Song of the day
+			handleShortCutForMenuIndex(0)
+			feedVC.pretappedPlusButton = true
+			break
+		case ShortcutIdentifier.PeopleSearch.type:
+			//Bring up People Search Screen
+			handleShortCutForMenuIndex(1)
+			break
+		case ShortcutIdentifier.Liked.type:
+			//Bring up Liked View
+			handleShortCutForMenuIndex(2)
+			break
+		case ShortcutIdentifier.Profile.type:
+			//Bring up Profile Screen (of current user)
+			profileVC.user = User.currentUser
+			handleShortCutForMenuIndex(-1)
+			break
```

Comment:
The break statements are redundant. Swift `switch` statements already break once a case is performed.